### ISSUE TITLE
Compilation error for taco generated code A(i,k) = B(i,j) * C(j,k)   (A,B,C are sparse)

### ISSUE THREAD

**Issue by tpops** [tensor-compiler/taco#307]
Hi, 

There is currently a compilation error for taco generated code:  

```sh
taco "A(i,k)=B(i,j)*C(j,k)" -f=A:ds:0,1 -f=B:ds:0,1 -f=C:ds:0,1 -write-source=taco_kernel.c -write-compute=taco_compute.c -write-assembly=taco_assembly.c
```

```cpp {.line-numbers}
int compute(taco_tensor_t *A, taco_tensor_t *B, taco_tensor_t *C, taco_tensor_t *A) {
  int A1_dimension = (int)(A->dimensions[0]);
  int B1_dimension = (int)(B->dimensions[0]);
  double* restrict B_vals = (double*)(B->vals);
  int C1_dimension = (int)(C->dimensions[0]);
  double* restrict C_vals = (double*)(C->vals);
  double* restrict A_vals = (double*)(A->vals);

  double A_val = A_vals[0];

  int32_t kA = 0;

  for (int32_t i = 0; i < B1_dimension; i++) {
    double* restrict A = 0;
    A = (double*)malloc(sizeof(double) * 42);
    for (int32_t pA = 0; pA < 42; pA++) {
      A[pA] = 0.0;
    }
    for (int32_t j = 0; j < C1_dimension; j++) {
      for (int32_t k = 0; k < ; k++) {
        double* restrict w = 0;
        w = (double*)malloc(sizeof(double) * 42);
        for (int32_t pw = 0; pw < 42; pw++) {
          w[pw] = 0.0;
        }
        w[kw] = w[kw] + B_vals[jB] * C_vals[kC];
        A[kA] = A_val;
        free(w);
        kA++;
      }
    }
    for (int32_t k = 0; k < ; k++) {
      double A_val0 = 0.0;
      A_val0 = [k];
      A[kA] = A_val0;
      kA++;
    }
    free(A);
  }

  return 0;
}
```

the line ` for (int32_t k = 0; k < ; k++) {` does not have an upper bound, thereby causing compilation error.

**Comment by stephenchouca** [comment]
This bug should have been fixed with the latest commit on master (99882ae). Can you see if that resolves your problem?

**Comment by tpops** [comment]
Yes it does thank you @stephenchouca
